bin/report

```diff
@@ -13,16 +13,14 @@ custom_packages=()
 custom_repositories=()
 
 while IFS= read -r line; do
-  if grep --silent --invert-match -e "^#" -e "^\s*$" <<< "${line}"; then
-    if grep --silent -e "^:repo:" <<< "${line}"; then
-      custom_repositories+=("${line//:repo:/}")
-    elif [[ $line == *deb ]]; then
-      custom_packages+=("${line}")
-    else
-      packages+=("${line}")
-    fi
+  if grep --silent -e "^:repo:" <<< "${line}"; then
+    custom_repositories+=("${line//:repo:/}")
+  elif [[ $line == *deb ]]; then
+    custom_packages+=("${line}")
+  else
+    packages+=("${line}")
   fi
-done < "${BUILD_DIR}/Aptfile"
+done < <(grep --invert-match -e "^#" -e "^\s*$" < "${BUILD_DIR}/Aptfile")
```

Comment:
Micro-optimization would be to remove the `<` so that `grep` gets the filename as an argument instead of on stdin.

Note that due to the process substitution, no error will occur if the `grep` somehow fails, or if `Aptfile` does not exist.

Reply:
I forgot you requested that in an earlier iteration @dzuelke.  Removed the `<` so now it'll use `grep --invert-match -e "^#" -e "^\s*$" "${BUILD_DIR}/Aptfile"`